lib/sinon/util/core/wrap-method.js

```diff
@@ -137,6 +137,7 @@ module.exports = function wrapMethod(object, property, method) {
         for (i = 0; i < types.length; i++) {
             mirrorProperties(methodDesc[types[i]], wrappedMethodDesc[types[i]]);
         }
+        methodDesc.configurable = true
```

Comment:
Do we need to store the original value, so it can be restored when the method is restored?

Reply:
The `restore()` function in this module already restores the original value. It only does so in the case of _own_ descriptors, otherwise it just deletes the spy/stub (thus removing the shadowing of the prototype).